components/dialog/dialog-mixin.js

```diff
@@ -177,13 +177,37 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		}
 	}
 
+	_findAutofocusElement(node) {
+		if (this._useNative) {
+			// Do not override native autofocus attribute implementation
+			return null;
+		}
+
+		const slot = node.querySelector('slot');
+		if (!slot) {
+			// We are in a confirm dialog, autofocus attribute will never be set
+			return null;
+		}
+
+		const content = slot.assignedElements({ flatten: true });
+
+		const searchOneElement = (el) => {
+			const foundElementOrNull = el.hasAttribute('autofocus') ? el : el.querySelector('[autofocus]');
+			return isFocusable(foundElementOrNull, false, false) ? foundElementOrNull : null;
```

Comment:
Using `isFocusable` enforces a `tabindex` of `1` or `0`, or semantic goodlist. This also means that custom elements will be required to add `tabindex` to use `autofocus`.

Reply:
That doesn't sound like something we want then? Maybe we just don't worry about `isFocusable` and if someone puts `autofocus` on something, we focus on it no matter what.

Reply:
Updated in https://github.com/BrightspaceUI/core/pull/2618/commits/3b75d6c506f9e906e61907dc733635e737af3a1c